server/src/main/java/com/lunchtime/service/implementation/OrderServiceImplementation.java

```diff
@@ -0,0 +1,66 @@
+package com.lunchtime.service.implementation;
+
+import com.lunchtime.models.Order;
+import com.lunchtime.models.Person;
+import com.lunchtime.models.RestaurantTable;
+import com.lunchtime.repository.OrderRepository;
+import com.lunchtime.service.OrderService;
+import com.lunchtime.service.PersonService;
+import com.lunchtime.service.RestaurantTableService;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class OrderServiceImplementation implements OrderService {
+
+    private final OrderRepository orderRepository;
+    private final RestaurantTableService restaurantTableService;
+    private final PersonService personService;
+
+    public OrderServiceImplementation(
+        OrderRepository orderRepository,
+        RestaurantTableService restaurantTableService,
+        PersonService personService
+    ) {
+        this.orderRepository = orderRepository;
+        this.restaurantTableService = restaurantTableService;
+        this.personService = personService;
+    }
+
+    public Order save(Order order) {
+        Optional<Person> person = personService.findById(order.getPerson().getId());
+        if (person.isPresent()) {
+            order.setPerson(person.get());
+        } else {
+            return null;
+        }
+
+        Optional<RestaurantTable> table = restaurantTableService.findById(order.getTable().getId());
+        if (table.isPresent()) {
+            order.setTable(table.get());
+        } else {
+            return null;
+        }
+        return orderRepository.save(order);
+    }
+
```

Comment:
Sorry, but I don't undersand lines 34-46 here. Could you explain this, please?

Reply:
I mean, why do you use setters if object is present?

Reply:
`Optional` is a class which helps to handle `NullPointerException`. `findById(id)` can don't find object in this case statement `else` will be used

I need to check if objects (`Person and RestaurantTable`) aren't `null` before set them

Reply:
It is Ok that you return null when person or restaurant are null. Otherwise, it seems that you find person from incoming order in line 34, and if it is present, you set it back to this order, isn't it?

Reply:
Yes

Reply:
So, do we need set it back when it exists?

Reply:
![image](https://user-images.githubusercontent.com/6152066/78360204-1a612980-75bf-11ea-8125-61f70b3b5d62.png)
![image](https://user-images.githubusercontent.com/6152066/78360389-690ec380-75bf-11ea-9006-8929e13eda87.png)

Yes

Reply:
Now I got it. Just passing id... 
Thank you!